File: Backend/marketplace_server/entity/user_entry_selection.py
from sqlalchemy import Column, BigInteger, Integer
from db import Base

class UserEntrySelection(Base):
    __tablename__ = 'user_entry_selections'
    user_id = Column(BigInteger, primary_key=True)
    entry_user_id = Column(BigInteger, primary_key=True)
    entry_node_id = Column(BigInteger, primary_key=True)
    selection_type = Column(Integer, nullable=False)
    created_at = Column(Integer)

File: serverless/rust/entity/user_entry_selection.rs
use sea_orm::entity::prelude::*;
use sqlx::FromRow;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, FromRow)]
#[sea_orm(table_name = "user_entry_selections")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub user_id: i64,
    #[sea_orm(primary_key)]
    pub entry_user_id: i64,
    #[sea_orm(primary_key)]
    pub entry_node_id: i64,
    pub selected_at: DateTime,
}

#[derive(Copy, Clone, Debug, EnumIter)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}